packages/hierarchies-react/src/presentation-hierarchies-react/itwinui/TreeNodeRenderer.tsx

```diff
@@ -8,49 +8,60 @@ import cx from "classnames";
 import { ComponentPropsWithoutRef, ReactElement } from "react";
 import { SvgFilter, SvgFilterHollow, SvgRemove } from "@itwin/itwinui-icons-react";
 import { Anchor, ButtonGroup, Flex, IconButton, ProgressRadial, Text, TreeNode } from "@itwin/itwinui-react";
-import { MAX_LIMIT_OVERRIDE } from "./internal/Utils";
-import { isPresentationHierarchyNode, PresentationHierarchyNode, PresentationTreeNode } from "./Types";
-import { useTree } from "./UseTree";
+import { MAX_LIMIT_OVERRIDE } from "../internal/Utils";
+import { isPresentationHierarchyNode, PresentationHierarchyNode } from "../TreeNode";
+import { useTree } from "../UseTree";
+import { RenderedTreeNode } from "./TreeRenderer";
 
 type TreeNodeProps = ComponentPropsWithoutRef<typeof TreeNode>;
 
 interface TreeNodeRendererOwnProps {
-  node: PresentationTreeNode;
-  onFilterClick: (nodeId: string | undefined) => void;
+  node: RenderedTreeNode;
+  onFilterClick?: (nodeId: string | undefined) => void;
   getIcon?: (node: PresentationHierarchyNode) => ReactElement | undefined;
-  onNodeClick: (nodeId: string, isSelected: boolean, event: React.MouseEvent<HTMLElement, MouseEvent>) => void;
-  onNodeKeyDown: (nodeId: string, isSelected: boolean, event: React.KeyboardEvent<HTMLElement>) => void;
+  onNodeClick?: (nodeId: string, isSelected: boolean, event: React.MouseEvent<HTMLElement, MouseEvent>) => void;
+  onNodeKeyDown?: (nodeId: string, isSelected: boolean, event: React.KeyboardEvent<HTMLElement>) => void;
   actionButtonsClassName?: string;
 }
 
-type TreeNodeRendererProps = Pick<ReturnType<typeof useTree>, "expandNode" | "setHierarchyLevelFilter" | "setHierarchyLevelLimit"> &
+type TreeNodeRendererProps = Pick<ReturnType<typeof useTree>, "expandNode"> &
+  Partial<Pick<ReturnType<typeof useTree>, "getHierarchyLevelDetails">> &
   Omit<TreeNodeProps, "label" | "onExpanded" | "onSelected" | "icon"> &
   TreeNodeRendererOwnProps;
 
-/** @beta */
+/**
+ * A component that renders `RenderedTreeNode` using the `TreeNode` component from `@itwin/itwinui-react`.
+ *
+ * @see `TreeRenderer`
+ * @see https://itwinui.bentley.com/docs/tree
+ * @beta
+ */
 export function TreeNodeRenderer({
   node,
   expandNode,
   getIcon,
-  setHierarchyLevelFilter,
   onFilterClick,
   onNodeClick,
   onNodeKeyDown,
-  setHierarchyLevelLimit,
   isSelected,
   isDisabled,
   actionButtonsClassName,
+  getHierarchyLevelDetails,
   ...nodeProps
 }: TreeNodeRendererProps) {
+  if ("type" in node && node.type === "ChildrenPlaceholder") {
```

Comment:
Why was this moved before `isPresentationHierarchyNode` check?

Reply:
because the "ChildrenPlaceholder" type of not is not a `PresentationHierarchyNode` anymore